src/client/screens/Playground/index.tsx

```diff
@@ -60,8 +66,11 @@ export default function Playground(): JSX.Element {
   };
 
   const socketReceivePenaltyRows = (rowsNumber: number) => {
-    console.log('PENALTY_ROWS receive, rowsNumber:', rowsNumber);
     // TODO: add rows to the bottom of matrix
+    // WHY STATE MATRIX IS ALWAYS EMPTY HERE????????
+    console.log('PENALTY_ROWS receive, rowsNumber:', rowsNumber, matrix);
+    // const newMatrix = addPenaltyRows(matrix, rowsNumber);
+    // setMatrix(newMatrix);
```

Comment:
I am stuck here, because on `socketReceivePenaltyRows` I always receive blank matrix. But current matrix in state is not blank

Reply:
it's because socketReceivePenaltyRows is used in useEffect that is only initialised in the beginning
![image](https://user-images.githubusercontent.com/6814254/99899506-27e03680-2caa-11eb-9120-a131220de9fd.png)

Reply:
use setMatrix trick like in useKeyEvent;
```
setMatrix((prevMatrix) => {
   console.log('PENALTY_ROWS receive, rowsNumber:', rowsNumber, prevMatrix);
}
```